airbyte-webapp/src/hooks/useRouter.tsx

```diff
@@ -1,39 +1,33 @@
 import queryString from "query-string";
 import { useMemo } from "react";
-import { useLocation, useNavigate, useParams, Location, To, NavigateOptions } from "react-router-dom";
+import { useLocation, useNavigate, useParams, To, NavigateOptions } from "react-router-dom";
 
-// eslint-disable-next-line @typescript-eslint/no-explicit-any
-function useRouter<T = any, P = any>(): {
-  query: T;
-  params: P;
-  pathname: string;
-  location: Location;
-  push(path: To, state?: NavigateOptions): void;
-  replace(path: To, state?: NavigateOptions): void;
-} {
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  const params: any = useParams();
+import { SortOrderEnum } from "../components/EntityTable/types";
+
+interface UseRouterQueryResult {
+  from?: string;
+  id?: string;
+  mode?: string;
+  oobCode?: string;
+  order?: SortOrderEnum;
+  sortBy?: string;
+}
+
+export const useRouterQuery = (): UseRouterQueryResult => {
```

Comment:
I think we should not keep this weird merge of params and queries around anymore. I think we should instead just have a very simpley `useQuery` hook for getting query parameters (without merging the path params in, since weirdly react-router doesn't have a hook for that) and useParams in all places we need the params. I've checked all current usages of this hook in this PR and left a comment which one they actually need (query or params), so I couldn't find any use-case for this mix of both.

Reply:
all fixed